### PR TITLE
Added some basic type checking & introduced more correct parsing of subtype

### LINKED ISSUE #333: subtype indication is not parsed

### PR DESCRIPTION
**Pull request by g0t00**
Closes #333

### CODE REVIEW

**Comment by Nik-Sch** [comment]
`objects.ts:232`: typeNames should also be `subtypeIndication`

File: lib/parser/expressionParser.ts [hunk 1]
```diff
@@ -158,7 +158,9 @@ export class ExpressionParser {
         // The collected tokens up to this point are then split and converted into OReferences and other objects
         const breakToken = breakTokens.find(token => token === (this.getNumToken()?.getLText() ?? ''));
         // Attributes are handled as one block, so check that the break token is not after a attribute ' mark.
-        if (breakToken && lastToken?.getLText() !== '\'') {
+        // To Identifiers after each other is a bit of a workaround. Normally this should not be possible. It happens if higher level parser splits a bit wrong.
```
Comment (Nik-Sch) [inline lib/parser/expressionParser.ts:161]:
```suggestion
        // Two Identifiers after each other is a bit of a workaround. Normally this should not be possible. It happens if higher level parser splits a bit wrong.
```

**Review by Nik-Sch — CHANGES_REQUESTED** [review]

File: lib/rules/ruleTypeResolved.ts [hunk 2]
```diff
@@ -88,11 +92,10 @@ export class RuleTypeResolved extends RuleBase implements IRule {
             let resolvedSubtype: string[] = [];
             if (implementsIHasDeclarations(root)) {
               for (const subtype of root.declarations) {
-                if (subtype instanceof O.OSubType && subtype.resolved && alias.includes(subtype.superType.nameToken.getLText())) {
+                if (subtype instanceof O.OSubType && subtype.subtypeIndication.resolutionIndication.length && alias.find(name => name == subtype.subtypeIndication.typeNames[0]?.nameToken?.getLText())) {
```
[the remark this answers is not in the document]
Comment (g0t00) [reply lib/rules/ruleTypeResolved.ts:95]: Not that i know of

File: lib/parser/subtypeIndicationParser.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        currentBucket.push(this.consumeToken());
+      }
+    }
+    buckets.push(currentBucket);
+
+    const subtypeIndication = new O.OSubtypeIndication(this.parent, startToken.range.copyWithNewEnd(this.getToken(-1, true).range));
+    if (buckets.length === 1) {
+      subtypeIndication.typeNames = new ExpressionParser(this.state, subtypeIndication, buckets[0]!).parse();
+    } else if (buckets.length === 2) {
+      // TODO: Find a more reliable way to determine which optional is set
```
[the remark this answers is not in the document]
Comment (g0t00) [reply lib/parser/subtypeIndicationParser.ts:56]: Look at the sentence below?

File: lib/rules/ruleTypeResolved.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (g0t00) [reply lib/rules/ruleTypeResolved.ts]: Okay I found one

**Review by Nik-Sch — APPROVED** [review]

**Pull request merged**